Add Drift Detector Benchmarker Prototype

yuji3w opened the pull request:
The objective of the drift detector benchmarker is to create an easy-to-use framework that benchmarks data drift detection methods. The benchmarked currently uses the NYC taxi dataset.

### Key features:
#### Created benchmarking framework, allowing easy testing of the following drift detectors created in this pr:
1. ClassifierDriftModel,
2. ClassifierUncertaintyModel,
3. CVMModel,
4. FETModel,
5. LearnedKernelModel,
6. MMDModel,
7. ModelWrapperInterface,
8. SpotTheDiffModel,
9. TabularModel,

#### Created graphing utility, allowing for plotting inter-week drift on a feature-wise level
#### Created accessor class and factory, allows for easy access to data frame data.
#### Created batch loader for loading large date ranges (especially greater than 1 year).

yuji3w committed:
Add change detection benchmarker prototype

shreyashankar commented on lib/data_clean_utils.py:
```diff
@@ -0,0 +1,24 @@
+"""
+def hard_coded_cleaning_steps_mutation(df: pd.DataFrame):
+    # Correct string type confusion in vendorid.
+    df.loc[df['vendorid'] == '1', 'vendorid'] = 1
```
```python
df['vendor_id'] = df['vendor_id'].astype(int)
df['payment_type'] = df['payment_type'].astype(int)
```

yuji3w replied:
Done. I converted to float first, then to int.

shreyashankar commented on model.ipynb:
```diff
@@ -0,0 +1,433 @@
+{
```
<p><code>tip_amount</code> provides leakage / information about the label <code>tip_percent_greater_15</code> -- so you might want to remove this feature</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/loglabs/ttb/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

yuji3w replied:
Done

shreyashankar commented on model.ipynb:
```diff
@@ -0,0 +1,433 @@
+{
```
<p>Typically comments go under the function definition!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/loglabs/ttb/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

yuji3w replied:
Done. Moved

shreyashankar commented on model.ipynb:
```diff
@@ -0,0 +1,433 @@
+{
```
<p>Can you save to one csv for all the methods? You can create a dafaframe with cols methodname, date range, p-value and then call <code>df.to_csv(..., index=False)</code></p>

 _Reply via <a href='https://app.reviewnb.com/loglabs/ttb/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

yuji3w replied:
Done.

yuji3w committed:
Add interpretability features + all models fully functional

1. Add f1, accuracy scores to CSV
2. Aggregate CSVs
3. Track runtimes for each method
4. All methods now ready for use

yuji3w committed:
Remove tip_amount from train labels

`tip_amount` leaks information to `tip_percent_greater_15`

shreyashankar commented on lib/data_clean_utils.py:
```diff
@@ -6,19 +6,10 @@
-    df.loc[df['vendorid'] == '2', 'vendorid'] = 2
-    df.loc[df['vendorid'] == '2.0', 'vendorid'] = 2
+    # Correct string type confusion in vendorid, payment_type
+    df['vendorid'] = df['vendorid'].astype(float).astype(int)
```
Why `astype(float)` before `astype(int)`?

yuji3w replied:
I needed to massage the type conversions because int("1.0") is invalid but int(float("1.0")) is valid.

shreyashankar commented on lib/model.py:
```diff
@@ -0,0 +1,180 @@
+        pass
+
+@ModelWrapperInterface.register
+class CVMModel:
```
Instead of using the register decorator, you can also just do:

```python
class MMDModel(ModelWrapperInterface):
...
```

yuji3w replied:
Done. This makes more sense since the register decorator seems to be used when the implementation isn't in the inheritance tree of the abc.

yuji3w committed:
Replace registration decorators with direct inheritance
